### PR TITLE
Adds an option to dm help

### PR DESCRIPTION
**Pull request by AutumnClove**
After creating your pull request, tick these boxes if they are applicable to you.

- [x] I have tested my changes against the `review` branch (the latest developmental version), and this pull request is targeting that branch as a base
- [x] I have tested my changes on Python 3.5/3.6

----

### Description
Adds an option to send the help menu directly to the user who called it. 
Useful for spam prevention. Additionally should prevent future problems with commands in dm's if more where to be added.

### CODE REVIEW

File: musicbot/bot.py [hunk 1]
```diff
@@ -1061,6 +1061,7 @@ def run(self):
             log.info("  Spotify integration: " + ['Disabled', 'Enabled'][self.config._spotify])
             log.info("  Legacy skip: " + ['Disabled', 'Enabled'][self.config.legacy_skip])
             log.info("  Leave non owners: " + ['Disabled', 'Enabled'][self.config.leavenonowners])
+            log.info("  Direct message help: " + ['Disabled', 'Enabled'][self.config.leavenonowners])
```
Comment (jayktaylor) [inline musicbot/bot.py]: Should not be `self.config.leavenonowners`

File: musicbot/bot.py [hunk 2]
```diff
@@ -1140,7 +1141,10 @@ def _gen_embed(self):
         if not self.is_all:
             desc += self.str.get('cmd-help-all', '\nOnly showing commands you can use, for a list of all commands, run `{}help all`').format(prefix)
 
-        return Response(desc, reply=True, delete_after=60)
+        if self.config.dmhelp:
+            await self.safe_send_message(message.author, desc, expire_in=60)
```
Comment (jayktaylor) [inline musicbot/bot.py]: It feels like we still should give acknowledgement in the user's channel that we've DM'd them?

**Review by jayktaylor — CHANGES_REQUESTED** [review]

File: musicbot/bot.py [hunk 3]
```diff
@@ -1140,7 +1141,11 @@ def _gen_embed(self):
...
+            return Response(":mailbox_with_mail:", reply=False, expire_in=15)
```
Comment (jayktaylor) [inline musicbot/bot.py]: Shouldn't this be `delete_after`?

**Review by jayktaylor — DISMISSED** [review]

**Review by AutumnClove — COMMENTED** [review]
Pls approve

**Comment by AutumnClove** [comment]
Closing because https://gist.github.com/Rapptz/4a2f62751b9600a31a0d3c78100287f1

**Pull request closed without merging**